tests/fv/fv_test.go

```diff
@@ -261,4 +260,150 @@ var _ = Describe("PolicyController", func() {
+					InterfaceName: "eth0",
+				},
+			}
+			_, err = calicoClient.WorkloadEndpoints().Create(&wep)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Definitively trigger a policy controller cache update by updating a pod's labels
+			idMetadata := api.WorkloadEndpointMetadata{
+				Name:         "eth0",
+				Node:         pod.Spec.NodeName,
+				Orchestrator: "k8s",
+				Workload:     "default." + podName,
+			}
+			pod.ObjectMeta.Labels["foo"] = "label2"
+			_, err = k8sClient.CoreV1().Pods("default").Update(&pod)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Wait for the policy controller to update the web with the new labels
+			Eventually(func() error {
+				w, err := calicoClient.WorkloadEndpoints().Get(idMetadata)
+				if err != nil {
+					return err
+				}
+				if w.Metadata.Labels["foo"] != "label2" {
+					return fmt.Errorf("%v should equal 'label2'", w.Metadata.Labels["foo"])
+				}
+				return nil
+			}, 3*time.Second).ShouldNot(HaveOccurred())
+
+			// Update the wep's ActiveInstanceID.
+			wep.Metadata.ActiveInstanceID = "aii2"
+			_, err = calicoClient.WorkloadEndpoints().Update(&wep)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Trigger a pod 'update' in policy controller by updating the pods labels
+			pod.Labels["foo"] = "label3"
+			_, err = k8sClient.CoreV1().Pods("default").Update(&pod)
+			Expect(err).NotTo(HaveOccurred())
+
+			var w *api.WorkloadEndpoint
+			Eventually(func() error {
+				var err error
+				w, err = calicoClient.WorkloadEndpoints().Get(idMetadata)
+				if err != nil {
+					return err
+				}
+				if w.Metadata.Labels["foo"] != "label3" {
+					return fmt.Errorf("%v should equal 'label3'", w.Metadata.Labels["foo"])
+				}
+				return nil
+			}, 3*time.Second).ShouldNot(HaveOccurred())
+
+			// Check that the policy controller updated its cache to the new ActiveInstanceID.
+			Expect(w.Metadata.ActiveInstanceID).To(Equal("aii2"))
+		})
+	})
+
+	It("labels are not updated when pod is not yet running", func() {
+		// Create a Pod
+		podName := "pod"
+		pod := v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: "default",
+				Labels: map[string]string{
+					"foo": "label1",
+				},
+			},
+			Spec: v1.PodSpec{
+				NodeName: "127.0.0.1",
+				Containers: []v1.Container{
+					v1.Container{
+						Name:    "container1",
+						Image:   "busybox",
+						Command: []string{"sleep", "3600"},
+					},
+				},
+			},
+		}
+		_, err := k8sClient.CoreV1().Pods("default").Create(&pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Update the pod's labels
+		pod.ObjectMeta.Labels["foo"] = "label2"
+		_, err = k8sClient.CoreV1().Pods("default").Update(&pod)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait for the policy controller to update the wep with the new labels
+		idMetadata := api.WorkloadEndpointMetadata{
+			Name:         "eth0",
+			Node:         pod.Spec.NodeName,
+			Orchestrator: "k8s",
+			Workload:     "default." + podName,
+		}
+		Consistently(func() error {
```

Comment:
Doesn't look like we're actually checking that the labels have changed in this test - is it done?

Reply:
Also, is `Consistently` the right thing to do here? Seems like it's probably just a case for an `Eventually`